src/sections/Meshery/Meshery-integrations/IntegrationsGrid.js

```diff
@@ -50,16 +51,27 @@ const IntegrationsGrid = ({ category, theme, count }) => {
 
   let [categoryNameList ,setcategoryNameList] = useState([{ id: -1,
     name: "All",
-    isSelected: true, },
+    isSelected: true, count: IntegrationList.length, },
   ...categoryNames.map((categoryName) => {
     return {
       id: categoryName,
       name: categoryName,
       isSelected: false,
+      count: categoryCount(categoryName),
     };
   })]
   );
 
+  function categoryCount(categoryName) {
+    let k = 0;
+    IntegrationList.map((integration) => {
+      if (integration.frontmatter.category === categoryName){
+        k++;
+      }
+    });
+    return k;
+  }
```

Comment:
Prefer arrow functions.
And, it would be better to use the `Array.reduce` method here instead of `map`.

Reply:
It is showing an error message if I try using reduce
`Error in function eval in ./src/sections/Meshery/Meshery-integrations/IntegrationsGrid.js:68
Cannot read properties of undefined (reading 'frontmatter')
`

Reply:
Usage of a different method shouldn't cause sich error. Probably, something wrong in the syntax/code that you are trying to implement 🤔.
If it doesn't work out, maybe you can share the code snippet and I can take a look as well.

Reply:
Ok, I think it is because of the initialValue that you are passing to the method. Add a null check inside the function definition. That should do the trick 😉

Reply:
```
const categoryCount = (categoryName) => {
    let k = 0;
    IntegrationList.map((integration) => {
      if (integration.frontmatter.category === categoryName){
        k++;
      }
    });
    return k;
  };

  let [categoryNameList ,setcategoryNameList] = useState([{ id: -1,
    name: "All",
    isSelected: true, count: IntegrationList.length, },
  ...categoryNames.map((categoryName) => {
    return {
      id: categoryName,
      name: categoryName,
      isSelected: false,
      count: categoryCount(categoryName),
    };
  })]
  );
```

Reply:
THis is the code. Okay let me try that out

Reply:
Where are you using the `reduce` method? 🤔

Reply:
```
const categoryCount = (categoryName) => {
    let k = 0;
    IntegrationList.reduce((integration) => {
      if (integration.frontmatter.category === categoryName){
        k++;
      }
    });
    return k;
  };
```

Reply:
It should be like this:
```
const categoryCount = (categoryName) => {
    return IntegrationList.reduce((count, integration) => {
      if (integration.frontmatter.category === categoryName){
        count += 1;
      }
    }, 0);
  };
```

Reply:
![integrations](https://user-images.githubusercontent.com/62856848/199061684-9c00aca1-765a-4890-a3ac-6202fc607945.png)

Reply:
The count is not coming

Reply:
Ahh...yes. Minor mistake.
```
  const categoryCount = (categoryName) => {
    return IntegrationList.reduce((count, integration) => {
      if (integration.frontmatter.category === categoryName){
        count += 1;
      }
      return count;
    }, 0);
  };
```
If the condition is false we should still be returning the existing value

Reply:
Ya it is working now. THanks a lot :)